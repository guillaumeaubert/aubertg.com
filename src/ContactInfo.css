#contact-info {
  display: block;
  position: absolute;
  top:0;
  right: 10px;
  z-index: 10;
  padding: 10px;
  box-shadow: 0 0 10px rgba(0,0,0,.5);
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  background-color: #0090ff;
  color: #fff;
  font-weight: 700;
  border-radius: 0px 0px 25px 25px;
}

#contact-info ul {
  list-style-type: none;
  margin: 0px auto;
  display: table;
  overflow: auto;
  padding: 0px 15px;
}

#contact-info ul li {
  float: left;
  margin-left: 16px;
}

#contact-info ul li:first-child {
  padding-top: 5px;
  margin-left: 0px;
}

#contact-info ul li img {
  display: block;
}
