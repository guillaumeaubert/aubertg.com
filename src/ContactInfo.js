// @flow strict

import React from 'react';
import githubIcon from './images/font-awesome_4-6-3_github_32_0_ffffff_none.png';
import linkedinIcon from './images/brandico_2014-04-07_linkedin-rect_32_0_ffffff_none.png';
import envelopeIcon from './images/font-awesome_4-6-3_envelope-square_32_0_ffffff_none.png';
import './ContactInfo.css';

type Props = {||};

class ContactInfo extends React.Component<Props> {
  render() {
    return (
      <div id="contact-info">
        <ul>
          <li>
            Find me on:
          </li>
          <li>
            <a href="https://github.com/guillaumeaubert/">
              <img src={githubIcon} alt="Github"/>
            </a>
          </li>
          <li>
            <a href="https://www.linkedin.com/in/aubertguillaume">
              <img src={linkedinIcon} alt="LinkedIn"/>
            </a>
          </li>
          <li>
            <a href="mailto:me@aubertg.com">
              <img src={envelopeIcon} alt="Email"/>
            </a>
          </li>
        </ul>
      </div>
    );
  }
}

export default ContactInfo;
