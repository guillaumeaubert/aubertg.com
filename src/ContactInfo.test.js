// @flow strict

import React from 'react';
import ContactInfo from './ContactInfo';
import renderer from 'react-test-renderer';

it('The component renders correctly', () => {
  const tree = renderer.create(
    <ContactInfo />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
