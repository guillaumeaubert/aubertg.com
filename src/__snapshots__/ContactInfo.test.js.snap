// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component renders correctly 1`] = `
<div
  id="contact-info"
>
  <ul>
    <li>
      Find me on:
    </li>
    <li>
      <a
        href="https://github.com/guillaumeaubert/"
      >
        <img
          alt="Github"
          src="font-awesome_4-6-3_github_32_0_ffffff_none.png"
        />
      </a>
    </li>
    <li>
      <a
        href="https://www.linkedin.com/in/aubertguillaume"
      >
        <img
          alt="LinkedIn"
          src="brandico_2014-04-07_linkedin-rect_32_0_ffffff_none.png"
        />
      </a>
    </li>
    <li>
      <a
        href="mailto:me@aubertg.com"
      >
        <img
          alt="Email"
          src="font-awesome_4-6-3_envelope-square_32_0_ffffff_none.png"
        />
      </a>
    </li>
  </ul>
</div>
`;
